include/zephyr/bluetooth/encrypted_ad_data.h

```diff
@@ -0,0 +1,77 @@
+/* Copyright (c) 2023 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include <zephyr/kernel.h>
+
+#define BT_EAD_AAD_SIZE	       1  /**< Additional Authenticated Data size in bytes */
+#define BT_EAD_RANDOMIZER_SIZE 5  /**< Randomizer size in bytes */
+#define BT_EAD_NONCE_SIZE      13 /**< nonce size in bytes */
+#define BT_EAD_KEY_SIZE	       16 /**< Key size in bytes */
+#define BT_EAD_IV_SIZE	       8  /**< Initialisation Vector size in bytes */
+#define BT_EAD_MIC_SIZE	       4  /**< MIC size in bytes */
+
+/* This value is used to set the directionBit of the CCM  nonce to the MSB of the Randomizer field
+ * (see Supplement to the Bluetooth Core   Specification v11, Part A 1.23.3)
+ */
+#define BT_EAD_RANDOMIZER_DIRECTION_BIT 7
+
+/* Fixed value used for the Additional Authenticated Data (see Supplement to the Bluetooth Core
+ * Specification v11, Part A 1.23.3)
+ */
+static uint8_t bt_ead_aad[] = {0xEA};
```

Comment:
Be careful about declaring variables in headers. Doing so requires header guards to prevent errors due to multiple definitions. And for sanity, any variables declared in header files must be static-const and should never have pointers to them. It's probably better to not declare variables in headers at all. (This will change when we start using C23 that has constexpr. :tada:)